test/integration/backward_compatible/parallel/serialization/compact/CompactPublicAPIsTest.js

```diff
@@ -0,0 +1,874 @@
+        queue = await client.getQueue(name);
+        set = await client.getSet(name);
+        topic = await client.getReliableTopic(name);
+        ringBuffer = await client.getRingbuffer(name);
+        compactSerializerUsed = false;
+    });
+
+    afterEach(async function () {
+        compactSerializerUsed.should.be.true;
+        sandbox.restore();
+        await map.destroy();
+        await nearCachedMap1.destroy();
+        await nearCachedMap2.destroy();
+        await multimap.destroy();
+        await replicatedMap.destroy();
+        await list.destroy();
+        await atomicReference.destroy();
+        await queue.destroy();
+        await set.destroy();
+        await topic.destroy();
+        await ringBuffer.destroy();
+        await testFactory.shutdownAllClients();
+    });
+
+    after(async function () {
+        await testFactory.shutdownAll();
+    });
+
+    const putToMapFromOtherClient = async (mapName, key, value) => {
+        const client = await testFactory.newHazelcastClientForParallelTests(clientConfig, member);
+        const map = await client.getMap(mapName);
+        await map.put(key, value);
+    };
+
+    const putToReplicatedMapFromOtherClient = async (replicatedMapName, key, value) => {
+        const client = await testFactory.newHazelcastClientForParallelTests(clientConfig, member);
+        const replicatedMap = await client.getReplicatedMap(replicatedMapName);
+        await replicatedMap.put(key, value);
+    };
+
+    const assertEntryEvent = async (events) => {
+        await TestUtil.assertTrueEventually(async () => {
+            events.length.should.be.equal(1);
+            const event = events[0];
+            event.key.should.be.deep.equal(INNER_INSTANCE);
+            event.value.should.be.deep.equal(OUTER_INSTANCE);
+            should.equal(event.oldValue, null);
+            should.equal(event.mergingValue, null);
+        });
+    };
+
+    describe('Map/NearCaches', function () {
+        it('aggregate', async function() {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                const result = await obj.aggregate(new CompactReturningAggregator());
+                result.should.be.deep.eq(OUTER_INSTANCE);
+            }
+        });
+
+        it('aggregateWithPredicate', async function() {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                const result = await obj.aggregateWithPredicate(new CompactReturningAggregator(), new CompactPredicate());
+                result.should.be.deep.eq(OUTER_INSTANCE);
+            }
+        });
+
+        it('containsKey', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.containsKey(car1);
+            }
+        });
+
+        it('containsValue', async function () {
+            for (const obj of [map, nearCachedMap1, nearCachedMap2]) {
+                await obj.containsValue(car1);
```

Comment:
minor: I believe using OUTER_INSTANCE would be much better as it contains nested compact. Same for others